torch/_inductor/pattern_matcher.py

```diff
@@ -1802,12 +1802,16 @@ def run_node(self, n: torch.fx.Node) -> Any:
 
 @torch.no_grad()
 def fwd_only(
-    fn: Callable[..., Any], args: Sequence[Any], *, run_dce: bool = True
+    fn: Callable[..., Any],
+    args: Sequence[Any],
+    *,
+    run_dce: bool = True,
+    decompositions: Mapping[torch._ops.OpOverload, Callable[..., Any]] = select_decomp_table(),
```

Comment:
This is going to run at function definition time. The decompositions can change - e.g. we have a option to not decompose rng ops to match exact eager numerics. 

Let's make this a callable `get_decomps_fn` and then run inside fwd_only

Reply:
Thanks! Something like this maybe?

```
    with enable_python_dispatcher():
        if decompositions is None:
            decompositions = select_decomp_table()

        gm = make_fx(fn, decompositions, tracing_mode="real")(*args)
```

Reply:
`decompositions = get_decomp_fn() if get_decomp_fn is not None else select_decomp_table()`

is what I was thinking

Reply:
Ah nice! Just out of curiousity, does making it a callable on function arg save memory footprint? Why do we prefer callable over just passing the dictionary?

Reply:
For same reason we are not binding the decomp at the function definition time - decomps can change based on configs, I think passing in function which comptues decomps will be less prone to accidentally ignoring current config option.